python/pyspark/ml/tests.py

```diff
@@ -163,6 +163,14 @@ def setParams(self, seed=None):
 
 class ParamTests(PySparkTestCase):
 
+    def test_copy_new_parent(self):
+        testParams = TestParams()
+        # Copying an instantiated param should fail
+        with self.assertRaises(ValueError):
+            testParams.maxIter._copy_new_parent(testParams)
+        # Copying a dummy param should succeed
+        TestParams.maxIter._copy_new_parent(testParams)
+
```

Comment:
Here we should also check the param's name, doc and parent are right just like what we do at ```test_param```.